src/spline_agent/context.py

```diff
@@ -18,66 +18,66 @@
 
 from spline_agent.datasources import DataSource
 from spline_agent.enums import WriteMode
-from spline_agent.exceptions import LineageTrackingContextNotInitialized
+from spline_agent.exceptions import LineageTrackingContextNotInitializedError
 from spline_agent.lineage_model import NameAndVersion
 
 logger = logging.getLogger(__name__)
 
 
 class LineageTrackingContext:
     def __init__(self):
-        self._name: Optional[str] = None
-        self._ins: List[DataSource] = []
-        self._out: Optional[DataSource] = None
-        self._write_mode: Optional[WriteMode] = None
-        self._system_info: Optional[NameAndVersion] = None
+        self.__name: Optional[str] = None
+        self.__ins: List[DataSource] = []
+        self.__out: Optional[DataSource] = None
+        self.__write_mode: Optional[WriteMode] = None
+        self.__system_info: Optional[NameAndVersion] = None
 
     @property
     def name(self) -> Optional[str]:
-        return self._name
+        return self.__name
 
     @name.setter
     def name(self, value: str):
         assert value is not None
-        if self._name is not None:
+        if self.__name is not None:
             logger.warning(f"Tracking context property 'name' is reassigned: "
-                           f"old value '{self._name}', new value '{value}'")
-        self._name = value
+                           f"old value '{self.__name}', new value '{value}'")
+        self.__name = value
 
     @property
     def inputs(self) -> tuple[DataSource, ...]:
```

Comment:
You are mixing an old typing style with the new one. Not a big deal though.

`tuple` (the standard type, ) vs `Tuple` (from typing module)

* https://peps.python.org/pep-0585/
* https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc

so if you are confident that you are always gonna be using Python 3.9 or higher, just use ones with lowercase and remove the imports from typing module

Reply:
fixed